Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -238,7 +238,7 @@ internal struct BinaryDecoder: Decoder {
                 if bodyBytes % itemSize != 0 || itemCount > UInt64(Int.max) {
                     throw BinaryDecodingError.truncated
                 }
-                value.reserveCapacity(value.count + Int(extendingOrTruncating: itemCount))
+                value.reserveCapacity(value.count + Int(truncatingIfNeeded: itemCount))
```

Comment:
This probably isn't right, and it likely means an update to https://github.com/apple/swift-protobuf/blob/master/Sources/SwiftProtobuf/MathUtils.swift#L42-L89 instead so the code will still compile with older Xcodes also.

Reply:
Yep.  Changing to the newest Swift 4 syntax is fine, but without updating MathUtils.swift, this entirely breaks Swift 3 compatibility.